src/interface.ts

```diff
@@ -15,37 +23,39 @@ enum ResolveError {
   ResolverTimeout = "ResolverTimeout"
 }
 
-interface Intent {
-  intent: IntentName;
-  context: Context;
-  /**
-   * Name of app to target for the Intent. Use if creating an explicit intent
-   * that bypasses resolver and goes directly to an app.
-   */
-  target?: AppIdentifier;
-  
-  /**
-   * Dispatches the intent with the Desktop Agent.
-   * 
-   * Accepts context data and target (if an explicit Intent) as optional args.
-   * Returns a Promise - resolving if the intent successfully results in launching an App.
-   * If the resolution errors, it returns an `Error` with a string from the `ResolveError` enumeration.
-   */
-  send(context: Context, target?: AppIdentifier): Promise<void>
+/**
+ * Specified whether the target app is already running or it can be launched by invokation.
+ */
+enum AppState {
```

Comment:
I'd like to better understand how we'd expect AppState to be used.  What are some examples of Apps that aren't launchable? Not installed?  Not the right user entitlements?  Wouldn't it be better for the desktop agent/app directory filter for this ahead of time in a way appropriate to the implementation?  Also, for Running state, how do we expect this information to change workflows?  For example: Hyperlinks don't distinguish between links to sites I already have open and ones I don't.   While there is load/start up time for any application, it generally doesn't inhibit workflows.